packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -1715,11 +1717,30 @@ class _SelectableFragment with Selectable, ChangeNotifier implements TextLayoutM
     return SelectionResult.none;
   }
 
+  SelectionResult _handleCollapseSelection(Offset globalPosition) {
+    _selectableContainsOriginWord = true;
+
+    final TextPosition position = paragraph.getPositionForOffset(paragraph.globalToLocal(globalPosition));
+    if (_positionIsWithinCurrentSelection(position) && _textSelectionStart != _textSelectionEnd) {
```

Comment:
Sorry about that, this block of code was pulled from `_handleSelectWord`s existing implementation so that case probably needs an adjustment or to be removed entirely.

Reply:
> This seems to be very similar to SelectWord, and probably SelectLine and SelectParagraph. What's the plan for expanding these functionality? An alternative is to use a single Selection event with different granularity

This will be addressed in the triple click gestures pr if we choose not to add a new selection event here.